app/src/main/java/com/developersbreach/composeactors/ui/screens/search/SearchViewModel.kt

```diff
@@ -21,20 +22,40 @@ class SearchViewModel @Inject constructor(
 
     var uiState by mutableStateOf(SearchUIState())
         private set
-
+    var searchType = SearchType.Actors
     /**
      * @param searchQuery user entered query in text field.
      * This function triggers everytime user makes query change.
      */
     fun performQuery(
         searchQuery: String
     ) {
+        if(searchType ==  SearchType.Actors) {
+            performQueryForActorsList(searchQuery)
+        } else {
+            performQueryForMoviesList(searchQuery)
+        }
+    }
+
+    private fun performQueryForMoviesList(searchQuery: String) {
```

Comment:
And here comes our core functional part for this feature.

```suggestion
fun performQuery(searchQuery: String) {
        viewModelScope.launch {
            when (searchType) {
                SearchType.Actors -> {
                    uiState = SearchUIState.ActorSearch(isSearchingResults = true)
                    val searchData = searchRepository.getSearchableActorsData(searchQuery)
                    uiState = (uiState as SearchUIState.ActorSearch).copy(
                        actorList = searchData,
                        isSearchingResults = false
                    )
                }
                SearchType.Movies -> {
                    uiState = SearchUIState.MovieSearch(isSearchingResults = true)
                    val searchData = searchRepository.getSearchableMoviesData(searchQuery)
                    uiState = (uiState as SearchUIState.MovieSearch).copy(
                        movieList = searchData,
                        isSearchingResults = false
                    )
                }
            }
        }
    }
```